src/directives/esriMap.js

```diff
@@ -56,6 +56,7 @@
             controller: function($scope, $element, $attrs) {
                 // get a reference to the controller
                 var self = this;
+                self.Basemaps = [];
```

Comment:
I'd prefer this to be called `_customBasemaps` and to provide an `addCustomBasemap()` function to the map controller's API.